File: Python/Gas Station.py
class Solution:
    # @param gas, a list of integers
    # @param cost, a list of integers
    # @return an integer
    def canCompleteCircuit(self, gas, cost):
        length = len(gas)
        cur = 0
        lack = 0
        index = 0
        for i in range(length):
            cur = cur + gas[i] - cost[i]
            if cur < 0:
                lack = lack + cur
                cur = 0
                index = i + 1
        return index if lack + cur >= 0 else -1

File: Java/src/Gas Station.java
public class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int cur = 0;
        int lack = 0;
        int index = 0;
        for (int i=0 ; i < gas.length ; i++){
            cur = cur + gas[i] - cost[i];
            if (cur < 0){
                lack = lack + cur;
                cur = 0;
                index = i + 1;
            }
        }
        if (lack + cur >= 0){
            return index;
        }
        else{
            return -1;
        }
    }
}